sycl/include/sycl/ext/intel/experimental/esimd/math.hpp

```diff
@@ -1592,6 +1592,37 @@ asin(T src0, int flag = saturation_off) {
   return Result[0];
 }
 
+// log
+
+template <int SZ>
+ESIMD_NODEBUG ESIMD_INLINE simd<float, SZ> log(simd<float, SZ> src0,
+                                               int flag = saturation_off) {
+  constexpr float ln2 = 0.693147f; // std::numbers::ln2_v<float> in c++20
```

Comment:
Nit: is this precision maximum float can reach? If so, could you please add a comment?

Reply:
For ln2 looks like it should be possible to add more digits after the dot. But for log2e (below) it seems to be the max precision already.

```
$ cat test.cpp 
#include <stdio.h>
int main() {
  float a1 = 0.693147180559945309417232121458176f;
  float b1 = 0.69314718f;
  int *ai1 = (int *)&a1;
  int *bi1 = (int *)&b1;
  printf("*ai1 == *bi1 %d\n", (*ai1 == *bi1));
  float a2 = 1.442695040888963407359924681001892f;
  float b2 = 1.442695f;
  int *ai2 = (int *)&a2;
  int *bi2 = (int *)&b2;
  printf("*ai2 == *bi2 %d\n", (*ai2 == *bi2));
  return 0;
}
$ g++ test.cpp; ./a.out 
*ai1 == *bi1 1
*ai2 == *bi2 1
$ 
```

I can actually add a double initializer and let compiler truncate it to float.